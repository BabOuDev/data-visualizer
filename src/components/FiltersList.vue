<template>
  <div class="filter-list" v-if="value.length">
    <span v-for="(filter, index) in value" :key="index">
      <label class="filter-label">
        {{filter.column}}: {{filter.values.join(' or ')}}
        <span class="remove-button" @click="removeFilter(index)">x</span>
      </label>
      <span v-if="index < value.length-1">and</span>
    </span>
  </div>
</template>

<script>

  export default {
    name: 'FiltersList',

    props: ['modelValue'],
    emits: ['update:modelValue'],

    computed: {
      value: {
        get() {
          return this.modelValue;
        },
        set(value) {
          this.$emit('update:modelValue', value);
        },
      },
    },
    methods: {
      removeFilter(index) {
        this.value.splice(index, 1);
      },
    },
  };
</script>

<style scoped>

.filter-list {
  display:block;
  user-select: none;
}

.filter-label{
  background-color: var(--color-2);
  color:white;
  padding: 2px 28px 2px 18px;
  margin: 2px 5px;
  position:relative;
  white-space: nowrap;
  display: inline-block;
}
.filter-label:hover .remove-button{
  display: block;
}
.remove-button{
  position: absolute;
  display: none;
  top: 4px;
  right: 5px;
  height: 18px;
  width: 18px;
  font-size: 15px;
  font-family: monospace;
  text-align: center;
  color: white;
  background-color: red;
  border-radius: 12px;
  cursor: pointer;
}

</style>
