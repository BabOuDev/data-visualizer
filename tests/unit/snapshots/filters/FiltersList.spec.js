import {shallowMount} from '@vue/test-utils';
import FiltersList from '@/components/filters/FiltersList.vue';

describe('FiltersList.vue', () => {
  it('should renders correctly', () => {
    const component = shallowMount(FiltersList, {
      props: {
        modelValue: [{column: 'ColName', values: ['A', 'B']}],
      },
    });
    expect(component.html()).toMatchSnapshot();
  });
});
