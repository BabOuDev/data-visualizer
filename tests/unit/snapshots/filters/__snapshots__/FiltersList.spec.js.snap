// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersList.vue should renders correctly 1`] = `
<div class="filter-list"><span><label class="filter-label">ColName: A or B <span class="remove-button">x</span></label>
  <!--v-if--></span>
</div>
`;
